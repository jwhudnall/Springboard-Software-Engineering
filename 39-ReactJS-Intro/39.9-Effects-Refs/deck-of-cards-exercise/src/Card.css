.Card {
  width: 225px;
  height: 314px;
  position: absolute;
}
