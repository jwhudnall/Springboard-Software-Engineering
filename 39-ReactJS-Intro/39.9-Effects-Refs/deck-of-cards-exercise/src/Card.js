import "./Card.css";

const Card = ({ img, rotation }) => {
  return (
    <div
      className='Card'
      style={{
        backgroundImage: `url(${img})`,
        transform: `translate(${rotation})`
      }}
    ></div>
  );
};

export default Card;
