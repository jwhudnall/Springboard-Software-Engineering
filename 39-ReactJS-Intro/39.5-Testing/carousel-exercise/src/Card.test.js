import { render, screen } from "@testing-library/react";

import Card from "./Card";

// Smoke test
it("renders", () => {
  render(<Card caption='sample-caption' src='link-to-somewhere' currNum={2} totalNum={3} />);
});

// Snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <Card caption='sample-caption' src='link-to-somewhere' currNum={2} totalNum={3} />
  );
  expect(asFragment()).toMatchSnapshot();
});
