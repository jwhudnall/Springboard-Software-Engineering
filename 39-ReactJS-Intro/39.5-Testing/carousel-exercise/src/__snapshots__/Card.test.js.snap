// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      sample-caption
    </h4>
    <img
      alt="sample-caption"
      class="Card-image"
      src="link-to-somewhere"
    />
    <small
      class="Card-small"
    >
      Image 2 of 3.
    </small>
  </div>
</DocumentFragment>
`;
